module/zfs/vdev.c

```diff
@@ -487,6 +488,19 @@ vdev_alloc(spa_t *spa, vdev_t **vdp, nvlist_t *nv, vdev_t *parent, uint_t id,
 
 	if (nvlist_lookup_string(nv, ZPOOL_CONFIG_PATH, &vd->vdev_path) == 0)
 		vd->vdev_path = spa_strdup(vd->vdev_path);
+
+	/*
+	 * ZPOOL_CONFIG_AUX_STATE = "external" means we previously forced a
+	 * fault on a vdev and want it to persist across imports (like with
+	 * zpool offline -f).
+	 */
+	nvlist_lookup_string(nv, ZPOOL_CONFIG_AUX_STATE, &tmp);
```

Comment:
It would be safer to additionally check the return code here instead of trusting that `nvlist_lookup_string` won't touch `&tmp`.